lib/vicadmin/validate.go

```diff
@@ -130,5 +134,23 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 		v.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
 	}
 
+	v.QueryDatastore(ctx, vch, sess)
 	return v
 }
+
+func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualContainerHostConfigSpec, sess *session.Session) {
+	var ds []mo.Datastore
+	refs := []types.ManagedObjectReference{sess.Datastore.Reference()}
```

Comment:
I got that datastore from the session variable, which I assume is (currently) the image datastore[0].  I'm revising to pull all datastores (including datastores which contain volumes)